src/FINDOLOGIC/Api/Config.php

```diff
@@ -165,7 +139,7 @@ public function setRequestTimeout($requestTimeout)
     public function getHttpClient()
     {
         if (!$this->httpClient) {
-            return new Client();
+            $this->httpClient = new Client();
```

Comment:
This seems intransparent, I would not expect a getter to change the state of the object. I can't think of a practical downside besides multi-threading race conditions (not a problem in PHP :grin:) but it still has a smell for me.

Consider returning `null` when no HTTP client is set.

Reply:
Ok but I prefer having a default, so nobody needs to explicitly set it, but it can be overriden. The default is set in the constructor.